src/python/WMCore/ReqMgr/Utils/Validation.py

```diff
@@ -26,6 +27,44 @@ def workqueue_stat_validation(request_args):
     stat_keys = ['total_jobs', 'input_lumis', 'input_events', 'input_num_files']
     return set(request_args.keys()) == set(stat_keys)
 
+def _validate_request_allowed_args(reqArgs, newReqArgs):
+    """
+    Compares two request configuration dictionaries and returns the difference
+    between them, but only at the first level (no recursive operations are attempted).
+    Returns the key/value pairs taken from the newReqArgs only if they are allowed
+    for the given request status.
+    :param reqArgs:    A dictionary with the current request definition
+    :param newReqArgs: A dictionary with user-provided request parameter changes
+    :return:           A dictionary reflecting the difference between the above two
+    NOTE:    This is asymmetrical operation/comparison, where newReqArs' values
+             are considered to take precedence but not the other way around. They
+             are in fact checked if they differ from the values at reqArgs and only those
+             items which differ are returned.
+    """
+    reqArgsDiff = {}
+    status = reqArgs["RequestStatus"]
+
+    # Checking all keys that differ:
+    reqArgsDiffKeys = []
+    for key in newReqArgs:
+        if key in reqArgs and newReqArgs[key] == reqArgs[key]:
+            continue
+        else:
+            reqArgsDiffKeys.append(key)
+
+    allowedKeys = deepcopy(get_modifiable_properties(status))
+
+    # NOTE: We need to explicitly add all stat keys during validation
+    #       They are needed for the workqueue_stat_validation() calls
+    ALLOWED_STAT_KEYS = ['total_jobs', 'input_lumis', 'input_events', 'input_num_files']
```

Comment:
Instead of proliferating these stat parameters, please create it under:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/ReqMgr/DataStructs/RequestStatus.py
and them import it here.

It would be great if you could update this line as well:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/ReqMgr/Utils/Validation.py#L26
such that we stop using this hardcoded variable, and use it from the RequestStatus module variables.

Reply:
In addition, if you do the separation that I suggested in Request.py, you won't need these extra checks over here.

Reply:
Importing the `ALLOWED_STAT_KEYS` is good. I'll change it, but we will still need the extra line  `allowedKeys.extend(ALLOWED_STAT_KEYS)` the other option would be to add it to the values returned  from `get_modifiable_properties(status)` irectly but this I am afraid may change behaviour for others who depend on this. Ergo this is why I used this mechanism to additionally extend the list here. rather than in `RequestSatatus`